internal/resource/config/password_credential_validator_resource.go

```diff
@@ -300,62 +218,19 @@ func (r *passwordCredentialValidatorsResource) Configure(_ context.Context, req
 
 func readPasswordCredentialValidatorsResponse(ctx context.Context, r *client.PasswordCredentialValidator, state *passwordCredentialValidatorsResourceModel, configurationFromPlan basetypes.ObjectValue) diag.Diagnostics {
 	var respDiags, diags diag.Diagnostics
-
 	state.Id = types.StringValue(r.Id)
 	state.CustomId = types.StringValue(r.Id)
 	state.Name = types.StringValue(r.Name)
-
-	// state.pluginDescriptorRef
-	pluginDescRef := r.GetPluginDescriptorRef()
-	state.PluginDescriptorRef = resourcelink.ToState(ctx, &pluginDescRef, &respDiags)
-
-	// state.parentRef
-	parentRef := r.GetParentRef()
-	state.ParentRef = resourcelink.ToState(ctx, &parentRef, &respDiags)
-
-	// state.Configuration
-	configurationAttrType := map[string]attr.Type{
-		"fields": basetypes.ListType{ElemType: types.ObjectType{AttrTypes: FieldAttrTypes()}},
-		"tables": basetypes.ListType{ElemType: types.ObjectType{AttrTypes: TableAttrTypes()}},
-	}
-
-	planFields := types.ListNull(types.ObjectType{AttrTypes: FieldAttrTypes()})
-	planTables := types.ListNull(types.ObjectType{AttrTypes: TableAttrTypes()})
-
-	planFieldsValue, ok := configurationFromPlan.Attributes()["fields"]
-	if ok {
-		planFields = planFieldsValue.(types.List)
-	}
-	planTablesValue, ok := configurationFromPlan.Attributes()["tables"]
-	if ok {
-		planTables = planTablesValue.(types.List)
-	}
-
-	fieldsAttrValue := ToFieldsListValue(r.Configuration.Fields, planFields, &diags)
-	tablesAttrValue := ToTablesListValue(r.Configuration.Tables, planTables, &diags)
-
-	configurationAttrValue := map[string]attr.Value{
-		"fields": fieldsAttrValue,
-		"tables": tablesAttrValue,
-	}
-	state.Configuration, diags = types.ObjectValue(configurationAttrType, configurationAttrValue)
+	state.PluginDescriptorRef = resourcelink.ToState(ctx, &r.PluginDescriptorRef, &respDiags)
```

Comment:
It looks like the ToState methods have different ways to handle diagnostics - these ones expect the diagnostics object as a parameter and others return a diagnostics object. We should probably try to make them consistent.